src/LinAlg/hiopMatrixDenseRowMajor.cpp

```diff
@@ -540,8 +542,11 @@ void hiopMatrixDenseRowMajor::timesMat_local(double beta, hiopMatrix& W_, double
   int M=X.n(), N=m_local_, K=X.m();
   int ldx=X.n(), ldm=n_local_, ldw=X.n();
 
-  double** XM=X.local_data(); double** WM=W.local_data();
-  DGEMM(&trans,&trans, &M,&N,&K, &alpha,XM[0],&ldx, this->M_[0],&ldm, &beta,WM[0],&ldw);
+  double* XM=X.local_data_const();
+  double* WM=W.local_data();
+  //DGEMM(&trans,&trans, &M,&N,&K, &alpha,XM[0],&ldx, this->M_[0],&ldm, &beta,WM[0],&ldw);
+
+  DGEMM(&trans,&trans, &M,&N,&K, &alpha,XM,&ldx, this->M_[0],&ldm, &beta,WM,&ldw);
```

Comment:
```suggestion
  DGEMM(&trans,&trans, &M,&N,&K, &alpha,XM,&ldx, this->local_data(),&ldm, &beta,WM,&ldw);
```
Can we replace reference to `M_` here?

Reply:
See line 588 below.

Reply:
left  `M_` untouched in rowmajor/cpu class